runtime/router/trie.go

```diff
@@ -140,8 +151,7 @@ func (t *tnode) set(path string, value http.Handler, lastKeyCharSlash, lastPathC
 	// is immediately after slash, e.g. "/:foo", "/x/:y". "/a:b" is not a colon wildcard segment.
 	var keyMatchIdx, pathMatchIdx int
 	for keyMatchIdx < keyLength && pathMatchIdx < pathLength {
-		if (t.key[keyMatchIdx] == ':' && lastKeyCharSlash) ||
-			(path[pathMatchIdx] == ':' && lastPathCharSlash) {
+		if t.isWildCardPattern(path, keyMatchIdx, pathMatchIdx, lastKeyCharSlash, lastPathCharSlash, false, isWhitelisted) {
```

Comment:
is the colonAsPattern hardcoded to false intentionally?

Reply:
Yes

Reply:
Fixed it